src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -0,0 +1,174 @@
+  private boolean openLoop;
+  private boolean inSecondStage;
+  
+  // Set proper constants later
+  private static final double STAGE_TRANSITION_HEIGHT = 25.0;
+  private static final double STAGE_1_CURRENT_LIMIT = 30.0;
+  private static final double STAGE_2_CURRENT_LIMIT = 35.0;
+  private static final double SLOW_ZONE_THRESHOLD = 2.0;
+  
+  public static Elevator getInstance() {
+    if (instance == null)
+      instance = new Elevator();
+    return instance;
+  }
+
+  private Elevator() {
+    ffd = new ElevatorFeedforward(0.0, Constants.Elevator.G_DOWN, 0.0);
+    ffu = new ElevatorFeedforward(0.0, Constants.Elevator.G_UP, 0.0);
+    
+    limitSwitch = new DigitalInput(Constants.Elevator.LIMIT_SWITCH_PORT);
+    leftMotor = Constants.Elevator.ELEVATOR_LEFT.createTalon();
+    rightMotor = Constants.Elevator.ELEVATOR_RIGHT.createTalon();
+    
+    leftMotor.setControl(new Follower(rightMotor.getDeviceID(), true));
+    
+    //creating config for software limit switch bec sushi lib doesnt handle
+    var config = new TalonFXConfiguration();
+    config.SoftwareLimitSwitch.ForwardSoftLimitEnable = true;
+    config.SoftwareLimitSwitch.ForwardSoftLimitThreshold = Constants.Elevator.MAX_HEIGHT;
+    config.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
+    config.SoftwareLimitSwitch.ReverseSoftLimitThreshold = 0;
+    rightMotor.getConfigurator().apply(config);
+    
+    resetElevator = false;
+    openLoop = false;
+    inSecondStage = false;
+    up = true;
+    
+    if (Constants.TUNING_MODE) {
+      pid = new PIDTuning("Elevator PID", Constants.Elevator.ELEVATOR_RIGHT.pid, Constants.TUNING_MODE);
+    }
+    
+    setpoint = new TunableNumber("Elevator Setpoint", 0, Constants.TUNING_MODE);
+  }
+
+  // moves elevator to defined ElevatorState position *last year specifically checked for elevator idle state
+  public Command changeState(ElevatorState state) {
+    return runOnce(() -> {
+      openLoop = false;
+      up = state.getPos() > rightMotor.getPosition().getValueAsDouble();
+      setpoint.setDefault(state.getPos());
+    }).andThen(new WaitUntilCommand(elevatorInPosition(state.getPos())))
+      .andThen(state == ElevatorState.IDLE ? resetElevator() : Commands.none());
+  }
+
+  // checks if elevator has reached target position
+  private BooleanSupplier elevatorInPosition(double targetPos) {
+    return () -> Math.abs(rightMotor.getPosition().getValueAsDouble() - targetPos) 
+      < Constants.Elevator.MAX_ERROR; // TODO: need to set max error properly
+  }
+
+  // uses limit switch to zero elevator
+  public Command resetElevator() {
+    return runOnce(() -> {
+      rightMotor.set(-0.1);
+      resetElevator = true;
+    }).andThen(Commands.waitUntil(this::elevatorAtBottom))
+      .andThen(runOnce(() -> {
+      rightMotor.set(0.0);
+      resetElevator = false;
+      rightMotor.setPosition(0.0); // zeroes
+    }));
+  }
+
+  // if pid isnt working
+  public Command runOpenLoopUp() {
```

Comment:
remove second stage checks, if we are running open loop we should be going pretty slow anyways. this should only be for debugging

Reply:
also just collapse these to a single runOpenLoop command